e2e/test/helpers/RetryOperationHelper.cs

```diff
@@ -38,7 +38,7 @@ public class RetryOperationHelper
                     await asyncOperation().ConfigureAwait(false);
                     return;
                 }
-                catch (Exception ex) when (!retryPolicy.ShouldRetry(counter, ex, out retryInterval))
+                catch (Exception ex) when (retryPolicy.ShouldRetry(counter, ex, out retryInterval))
```

Comment:
Exception should be retried only if the retry policy states so. Else the exception should be thrown back to the caller test.